src/Microsoft.DotNet.Tools.Repl.Csi/Program.cs

```diff
@@ -14,26 +19,148 @@ public static int Main(string[] args)
             return app.Execute(args);
         }
 
-        private static int Run(string scriptOpt)
+        private static ProjectContext GetProjectContext(IEnumerable<string> targetFrameworks, string projectPath)
+        {
+            var contexts = ProjectContext.CreateContextForEachFramework(projectPath);
+            var context = contexts.First();
+
+            if (targetFrameworks.Any())
+            {
+                var framework = NuGetFramework.Parse(targetFrameworks.First());
+                context = contexts.FirstOrDefault(c => c.TargetFramework.Equals(framework));
+            }
+
+            return context;
+        }
+
+        private static CommandResult CompileProject(ProjectContext projectContext, string configuration, out string tempOutputDir)
+        {
+            tempOutputDir = Path.Combine(projectContext.ProjectDirectory, "bin", ".dotnetrepl", Guid.NewGuid().ToString("N"));
+
+            Reporter.Output.WriteLine($"Compiling {projectContext.RootProject.Identity.Name.Yellow()} for {projectContext.TargetFramework.DotNetFrameworkName.Yellow()} to use with the {"C# REPL".Yellow()} environment.");
+
+            return Command.Create($"dotnet-compile", $"--output \"{tempOutputDir}\" --temp-output \"{tempOutputDir}\" --framework \"{projectContext.TargetFramework}\" --configuration \"{configuration}\" \"{projectContext.ProjectFile.ProjectDirectory}\"")
+                                .ForwardStdOut(onlyIfVerbose: true)
+                                .ForwardStdErr()
+                                .Execute();
+        }
+
+        private static IEnumerable<string> GetRuntimeDependencies(ProjectContext projectContext, string buildConfiguration)
+        {
+            var runtimeDependencies = new HashSet<string>();
+
+            var projectExporter = projectContext.CreateExporter(buildConfiguration);
+            var projectDependencies = projectExporter.GetDependencies();
+
+            foreach (var projectDependency in projectDependencies)
+            {
+                var runtimeAssemblies = projectDependency.RuntimeAssemblies;
+
+                foreach (var runtimeAssembly in runtimeAssemblies)
+                {
+                    var runtimeAssemblyPath = runtimeAssembly.ResolvedPath;
+                    runtimeDependencies.Add(runtimeAssemblyPath);
+                }
+            }
+
+            return runtimeDependencies;
+        }
+
+        private static string CreateResponseFile(ProjectContext projectContext, string buildConfiguration, string tempOutputDir)
+        {
+            var outputFileName = projectContext.ProjectFile.Name;
+            var outputFilePath = Path.Combine(tempOutputDir, $"{outputFileName}{Constants.DynamicLibSuffix}");
+            var projectResponseFilePath = Path.Combine(tempOutputDir, $"dotnet-repl.{outputFileName}{Constants.ResponseFileSuffix}");
+
+            var runtimeDependencies = GetRuntimeDependencies(projectContext, buildConfiguration);
```

Comment:
Perhaps @agocke could provide some better insight. I provided a comment detailing that it collects the full list of runtime dependencies and, at the very least, it is required for library projects where the dependencies don't appear to be placed in the output folder. Part of it may also be that the dependencies can/will change depending on what framework you are targeting.